vertical-pod-autoscaler/hack/vpa-process-yamls.sh

```diff
@@ -40,7 +40,19 @@ if [ $# -gt 2 ]; then
 fi
 
 ACTION=$1
-COMPONENTS="vpa-v1-crd-gen vpa-rbac updater-deployment recommender-deployment admission-controller-deployment"
+COMPONENTS="vpa-v1-crd-gen vpa-rbac updater-deployment recommender-deployment admission-controller-deployment recommender-externalmetrics-deployment"
+TEST_COMPONENTS=(recommender-externalmetrics-deployment)
+
+function script_path {
+  # Regular components have deployment yaml files under /deploy/.  But some components only have
+  # test deployment yaml files that are under hack/e2e.
+  if printf '%s\0' "${TEST_COMPONENTS[@]}" | grep -Fqxz -- $1; then
```

Comment:
I don't understand this condition. Could you explain it (or rewrite it to make it easier to understans)